docs/lesson-plans/week-1.md

```diff
@@ -0,0 +1,726 @@
+people can work on the same thing at the same time while providing a way to combine their changes together).
+- GitHub is a website that hosts git 'repositories' for many major companies and open-source projects (meaning their 
+code is open to anyone to view/edit/use). It provides useful features, like allowing people to review changes before
+they're accepted and being a remote server that people can use for storing their code and collaborating with 
+others.
+- Repositories (or 'repos') represent an individual 'project' in git.
+- Cloning is when you download a repository from an external host (such as GitHub) to your computer.
+- Checkout is a command in git that lets you switch the current code to a different branch.
+- Branches are the way that a different history is created for a specific feature or bug that is being worked on.
+They allow you to split off from a specific point and make isolated changes without having to worry that you're 
+changing the code you split off from. 
+![git-branches](./img/git-branch.svg)
+*Git Branches/Atlassian*
+
+### Installing IntelliJ
+
+The way that you'll be interacting with a lot of the code for this project will be through IntelliJ.
+Go to the [install link](https://www.jetbrains.com/idea/download/) and download the
+'Ultimate' edition. Since you're a student, you aren't limited to the 30-day trial that would otherwise
+be applied.
+
+Once you've downloaded and installed it (you'll probably need to create an account with your student 
+email), open it so that the welcome page shows up.
+
+>Note: some people prefer to use text editors like [Microsoft's VSCode](https://code.visualstudio.com/) or
+>[GitHub's Atom](https://atom.io/) instead of IntelliJ for working on the frontend. We'll be focusing
+>on using IntelliJ for all of this, but feel free to try those out if you're interested.
+
+### Cloning the repositories with Git
+
+Now that you've gotten IntelliJ set up, we'll get to cloning the repositories on your computer.
+
+>Note: you may need a GitHub account to do this. If you encounter an error trying to clone it,
+>then then just sign up and and retry the steps after logging into GitHub in IntelliJ.
+
+Inside of IntelliJ, go to File > New... > Project From Version Control..., make sure the 
+`Version Control` box is set to `Git`, paste in the following links, and choose where you 
+want them to be downloaded to. 
+
+>Note: if you're on Windows, you may also need to install git (it should give you a button to do so)
+>to be able to clone the projects.
+
+The links you'll want to use are:
+- Frontend: `https://github.com/Code-4-Community/jumpstart-frontend.git`
+- Backend: `https://github.com/Code-4-Community/jumpstart-backend.git`
+
+After the projects are open, you'll want to view the starter code by entering the following command into the terminal
+at the bottom of your IntelliJ window for both the frontend and backend projects.
+
+```
+git checkout starter-code
```

Comment:
Wouldn't they have to pull this branch from the remote first? Or does IntelliJ take care of that for them somewhere?

Reply:
I think git pulls references to all branches present when it clones the project, so I've always just been able to do that.